ТЗ 14, проект ShareIt. Для объектов Item и User были созданы:

Vladyak1 opened the pull request:
- DTO;
- контроллеры;
- методы Service и Repository;
- добавлены классы для исключений.

Кроме того, на просторах GitHub нашёл другой репозиторий и решил позаимствовать применение mapStruct для мапперов и класс-обертку для методов контроллера ResponseEntity

Vladyak1 committed:
ТЗ 14, проект ShareIt. Для объектов Item и User были созданы:
- DTO;
- контроллеры;
- методы Service и Repository;
- добавлены классы для исключений.

Кроме того, на просторах GitHub нашёл другой репозиторий и решил позаимствовать применение mapStruct для мапперов и класс-обертку для методов контроллера ResponseEntity

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/booking/BookingStatus.java:
```diff
@@ -0,0 +1,10 @@
+
+public enum BookingStatus {
+
+    REQUESTED, // запрос на бронирование отправлен пользователем
```
Комментарии здесь лишние

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/handler/ErrorHandler.java:
```diff
@@ -0,0 +1,77 @@
+
+@Slf4j
+@RestControllerAdvice
+public class ErrorHandler {
```
Отличная обработка ошибок

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/item/ItemRepository.java:
```diff
@@ -0,0 +1,20 @@
+
+public interface ItemRepository {
+
+    Optional<Item> create(Item item);
```
Для чего вам на создание возвращать обертку Optional ? На создание можно или ничего не возвращать или возвращать созданную сущность

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/item/ItemRepository.java:
```diff
@@ -0,0 +1,20 @@
+
+    List<Item> getAllByOwner(long ownerId);
+
+    Optional<Item> get(long id);
```
Считаю что здесь все обертки Optional - лишние, всем клиентам этого интерфейса придется эту обертку Optional обрабатывать, а лучше разобраться с ней сразу же внутри реализации интерфейса, а уже пусть все остальные (кто использует методы интерфейса) работают с конечным результатом (если нужен, иногда достаточно void)

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/item/ItemRepositoryInMemory.java:
```diff
@@ -0,0 +1,63 @@
+@Repository
+public class ItemRepositoryInMemory implements ItemRepository {
+
+    private final HashMap<Long, Item> items;
```
Тип данных для реализации HashMap - выбирайте Map 
То же самое List для ArrayList или LinkedList 
Вот пару ссылок с объявнением

https://softwareengineering.stackexchange.com/a/225677
https://stackoverflow.com/a/20793630/12210331

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/item/ItemRepositoryInMemory.java:
```diff
@@ -0,0 +1,63 @@
+        if (text.isEmpty() || text.isBlank()) {
+            return Collections.emptyList();
+        }
+        return items.values().stream()
```
Здорово работаете со стримами

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/user/UserRepository.java:
```diff
@@ -0,0 +1,19 @@
+import java.util.List;
+import java.util.Optional;
+
+public interface UserRepository {
```
Здесь те же комменты, что и к ItemRepository

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/validator/Marker.java:
```diff
@@ -0,0 +1,13 @@
+ * Маркерные интерфейсы для аннотаций
+ */
+
+public interface Marker {
```
Желательное давать более говорящее название для интерфейса

Vladyak1 replied:
Решил написать так, чтобы было удобно использовать при вставке в валидацию, но не настаиваю на этом варианте. Как лучше его переименовать?

VladimirTitov4 replied:
Это самый сложный вопрос) Обычно интерфейсы называют глаголом, Swimmable, Flyable, т.е. способный что-то делать, у вас здесь интерфейс - маркер для валидаций, можно сделать Checkable, verifiable

VladimirTitov4 commented on src/main/java/ru/practicum/shareit/user/mapper/UserMapper.java:
```diff
@@ -0,0 +1,14 @@
+import ru.practicum.shareit.user.User;
+import ru.practicum.shareit.user.dto.UserDto;
+
+@Mapper(componentModel = "spring")
```
Отлично, что используете Mapstruct

Вместо строки "spring" можно использовать константу 

org.mapstruct.MappingConstants.ComponentModel.SPRING

Vladyak1 committed:
ТЗ14, первые исправления:
- убраны исчерпывающие комментарии;
- слой repository сам проводит проверку наличия значений и возвращает сам объект, а не Optional;
- @Mapper теперь использует константу, а не строчное значение.

Vladyak1 committed:
Исправление под тесты Postman:
- POST запросы теперь возвращают статус OK, а не CREATED.

Vladyak1 commented:
Тесты не проходят, не пойму смысл ошибки, к примеру: "id" must be 3: expected 3 to deeply equal '3'

Vladyak1 committed:
@Mapper теперь использует константу вместо строкового значения

Vladyak1 committed:
interface Marker переименован в ValidateWhile
